cast-dssspell.sh

```diff
@@ -7,25 +7,42 @@ CHANGELOG=0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F
 MCD_ADM=$(seth call "$CHANGELOG" 'getAddress(bytes32)(address)' "$(seth --to-bytes32 "$(seth --from-ascii "MCD_ADM")")")
 MCD_GOV=$(seth call "$CHANGELOG" 'getAddress(bytes32)(address)' "$(seth --to-bytes32 "$(seth --from-ascii "MCD_GOV")")")
 MCD_IOU=$(seth call "$CHANGELOG" 'getAddress(bytes32)(address)' "$(seth --to-bytes32 "$(seth --from-ascii "MCD_IOU")")")
-TEN_THOUSAND=$(seth --to-wei 10000 ether)
+BALANCE=$(seth call "$MCD_GOV" 'balanceOf(address)(uint256)' "$ETH_FROM")
+HAT=$(seth call "$MCD_ADM" 'hat()(address)')
+APPROVALS=$(seth call "$MCD_ADM" 'approvals(address)(uint256)' "$HAT")
+DEPOSITS=$(seth call "$MCD_ADM" 'deposits(address)(uint256)' "$ETH_FROM")
+CONTI=1
 
-if [[ -z "$1" ]]; then
+if [[ -z "$1" ]];
+then
     echo "Please specify the Goerli Spell Address"
-else
-    seth send "$MCD_GOV" 'approve(address,uint256)' "$MCD_ADM" "$TEN_THOUSAND"
-    seth send "$MCD_ADM" 'lock(uint256)' "$TEN_THOUSAND"
-    
+elif [[ "$APPROVALS" -ge "$DEPOSITS" ]];
+then
+    DELTA=$((APPROVALS - DEPOSITS))
+    LOCK_AMOUNT=$((DELTA + CONTI))
+
+    [[ "$BALANCE" -ge "$LOCK_AMOUNT" ]] || { echo "$ETH_FROM: Insufficient MKR Balance"; exit 1; }
+
+    seth send "$MCD_GOV" 'approve(address,uint256)' "$MCD_ADM" "$LOCK_AMOUNT"
+    seth send "$MCD_ADM" 'lock(uint256)' "$LOCK_AMOUNT"
+
+    DEPOSITS=$(seth call "$MCD_ADM" 'deposits(address)(uint256)' "$ETH_FROM")
+
+    else
```

Comment:
i'm probably missing it in my bash reading, but where is the `if` for this `else`?

Reply:
should be the `elif`

Reply:
should it be indented one less then?

also, so the logic then is if approvals is greater than deposits we add to our deposits but don't vote and cast the spell?

Reply:
fixed indentation in 917c68c

Reply:
double checking

Reply:
fixed in 1c82345